### ISSUE TITLE
About Lazy Loading 

### ISSUE THREAD

**Issue by pascalf22** [Niresh12495/LazyAdsense#1]
Hello Niresh !

Thanks about it ! I have a question... does it respect Google Policies?
If i want not to lazy load but only differ all ads after the load page.. to not affect the PageSpeed... Is it the same ?

Thanks !
Pascal

**Comment by Niresh12495** [comment]
No,
All it does is inject the adsense adcode when the element gets visible
For an example when you convert typical adsense code to lazyadsense code
```
<div class="lazyadsense"
     style="display:block"
     data-ad-client="ca-pub-XXXXXX"
     data-ad-slot="YYYY"
     data-ad-format="auto"
     data-full-width-responsive="true"></div>
```

This particular element is converted into Adsense code and injected in place when the element becomes visible.
I have used this code in websites for morethan 6 months and had no issues with Adsense at all.

But if you want to defer load all the Adsense ads
use the following code 
```
<script>
function downloadJSAtOnload() {
    var element = document.createElement("script");
    element.src = "https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js?client=ca-pub-XXXXXXXXXXXXXXXX";
    element.async = true;
    element.setAttribute('crossorigin', 'anonymous');
    document.body.appendChild(element);
}
if (window.addEventListener)
    window.addEventListener("load", downloadJSAtOnload, false);
else if (window.attachEvent)
    window.attachEvent("onload", downloadJSAtOnload);
else window.onload = downloadJSAtOnload;
</script>
```

You can also try [LazyHTML](https://github.com/Niresh12495/LazyHTML) to Lazy Load Adsense, Which does not need to modify your Adsense adcode.

**Comment by pascalf22** [comment]
Hi @Niresh12495,

Thanks a lot for your quick response ! I really appreciated ! 

Thank you for all your explanation ! I will have a look at lazyadsens code and LazyHTML !

One last question please... just to be sure because it is important for me..
if I use the defer one:
```
<script>
function downloadJSAtOnload() {
    var element = document.createElement("script");
    element.src = "https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js?client=ca-pub-XXXXXXXXXXXXXXXX";
    element.async = true;
    element.setAttribute('crossorigin', 'anonymous');
    document.body.appendChild(element);
}
if (window.addEventListener)
    window.addEventListener("load", downloadJSAtOnload, false);
else if (window.attachEvent)
    window.attachEvent("onload", downloadJSAtOnload);
else window.onload = downloadJSAtOnload;
</script>
```

Do you think that it is against Google Policies ?
I know you tried lazy load for 6 months.. but for the defer ?

Because the best way for me... is all ads load... after the page load to not affect my PageSpeed !
Thanks a lot !

I really appreciate it :)

Have a great day ! 

Pascal

**Comment by Niresh12495** [comment]
>  Do you think that it is against Google Policies ?

No, you're just loading the adsbygoogle.js into dom on condition.
Defer loading might affect your revenue a little bit, as the ads are loaded once the Page is rendered.

**Comment by pascalf22** [comment]
Thanks again for your response ! I appreciated a lot ! I will try lazyadsense code and LazyHTML!
Thanks again !
Have a great day ! 
Pascal
